contracts/0.4.24/oracle/Algorithm.sol

```diff
@@ -11,55 +11,62 @@ library Algorithm {
     using SafeMath for uint256;
 
     /**
-      * Computes mode of a non-empty array, if array is unimodal.
-      * Low gas cost.
+      * @dev Finds the first element in the given array |data| that occures
+      * more that |quorum| times. Low gas cost for majority case. Returns the
+      * discovered element or 0 if no element is frequent enough.
       */
-    function mode(uint256[] data) internal pure returns (bool, uint256) {
-
+    function frequent(uint256[] data, uint256 quorum) internal pure returns (uint256) {
```

Comment:
Maybe return (bool, uint256)? And return (false, 0) if there's no quorum. In practice 0 is unused and there will be no overlap, but I think that clarity outweighs minor gas economy in this case.

Reply:
Yes, I thought the same indeed, epochId cannot be 0. But OK, I'll switch to (bool, val) pair.

Reply:
oracles don't report epochId though, they report balances